applications/GeoMechanicsApplication/custom_processes/find_neighbour_elements_of_conditions_process.cpp

```diff
@@ -214,40 +176,103 @@ void FindNeighbourElementsOfConditionsProcess::Execute()
-                                pCondition->Set(VISITED,true);
-                                pCondition->SetValue(NEIGHBOUR_ELEMENTS, VectorOfNeighbours);
-                            }
-                        }
+                        CheckForMultipleConditionsOnElement(FacesMap, itFace, itElem);
                     }
                 }
             }
         }
     }
 
+
+    //check that all of the conditions belong to at least an element.
+    all_conditions_visited = CheckIfAllConditionsAreVisited();
+
+    if (all_conditions_visited) {
+        // if all conditions are found, no need for further checks:
+        return;
+    }
+
+    // check 1D elements, note that this has to happen after procedures to find 2 and 3d neighbours are alredy performed, such that 1D elements are only added
+    // as neighbours when the condition is not neighbouring 2D or 3D elements
+    this->CheckIf1DElementIsNeighbour(FacesMap);
+    
+    
     //check that all of the conditions belong to at least an element. Throw an error otherwise (this is particularly useful in mpi)
-    AllVisited = true;
+    all_conditions_visited = true;
     for (auto& rCond : mrModelPart.Conditions()) {
         if (rCond.IsNot(VISITED)) {
-            AllVisited = false;
+            all_conditions_visited = false;
             KRATOS_INFO("Condition without any corresponding element, ID ") << rCond.Id() << std::endl;
         }
     }
 
-    KRATOS_ERROR_IF_NOT(AllVisited) << "Some conditions found without any corresponding element" << std::endl;
+    KRATOS_ERROR_IF_NOT(all_conditions_visited) << "Some conditions found without any corresponding element" << std::endl;
 
     KRATOS_CATCH("")
 }
 
+bool FindNeighbourElementsOfConditionsProcess::CheckIfAllConditionsAreVisited() const
+{
+    for (auto& r_cond : mrModelPart.Conditions()) {
+        if (r_cond.IsNot(VISITED)) {
+            return false;
+        }
+    }
+    return true;
+}
+
+void FindNeighbourElementsOfConditionsProcess::CheckIf1DElementIsNeighbour(hashmap& rFacesMap)
+{
+    // Now loop over all elements and check if one of the faces is in the "FacesMap"
+    for (auto itElem = mrModelPart.ElementsBegin(); itElem != mrModelPart.ElementsEnd(); ++itElem) {
+        const auto& r_geometry_element = itElem->GetGeometry();
+
+	    // for 1D elements, the edge geometry is the same as the element geometry 
+	    if (r_geometry_element.LocalSpaceDimension() == 1)
+	    {
+            const auto rBoundaryGeometries = PointerVector(r_geometry_element.GenerateEdges());
+
+            for (IndexType iFace = 0; iFace < rBoundaryGeometries.size(); ++iFace) {
+                DenseVector<int> FaceIds(rBoundaryGeometries[iFace].size());
+
+                // get edge ids
+                for (IndexType iNode = 0; iNode < FaceIds.size(); ++iNode) {
+                    FaceIds[iNode] = rBoundaryGeometries[iFace][iNode].Id();
+                }
```

Comment:
I believe this is a transformation:
```suggestion
                const auto& rNodes = rBoundaryGeometries[iFace];
                std::transform(rNodes.begin(), rNodes.end(), FaceIds.begin(),
                               [](const auto& rNode){ return rNode.Id(); });
```
And once again: please check this suggestion carefully before incorporating it. Thanks.